### PR TITLE
Enable cr50 tpm drivers

### PR DESCRIPTION
**Pull request by 10ne1**
HI,

This is the kernelci-core counterpart to the [bootrr PR #12](https://github.com/kernelci/bootrr/pull/12).

[Here is an example run](https://lava.collabora.co.uk/results/testcase/170445026) of the bootrr test with these configs and tpm2-tools on my dev setup.

### CODE REVIEW

**Comment by eballetbo** [comment]
See my comment in bootrr PR #12 about tpm2-tools. I think is not needed.

**Comment by 10ne1** [comment]
@eballetbo I responded to [bootrr PR #12](https://github.com/kernelci/bootrr/pull/12). I think we need to keep tpm2-tools because not all drivers expose the same /sys/class/tpm info and by using tpm2-tols we use a standard open source userspace software stack to sanity check any driver on any devices (including non-Chromebooks) by validating communication with the /dev/tpm device directly.

**Comment by eballetbo** [comment]
LGTM

**Comment by gctucker** [comment]
> The drivers expose /dev/tpm devices which will then be used by tpm2-tools in Debian images or a custom TPM userspace stack in ChromeOS images exercisable via the TAST test suite.

Are all the config options useful for `tpm2-tools` or are some only required for TAST?

Anything that is solely to enable TAST to run should be used only on the chromeos.kernelci.org instance to keep linux.kernelci.org as close to mainline defconfigs as possible.

File: config/core/rootfs-configs.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ rootfs_configs:
       - mipsel
     extra_packages:
       - isc-dhcp-client
+      - tpm2-tools
```
[the remark this answers is not in the document]
Comment (gctucker) [reply config/core/rootfs-configs.yaml:30]: Actually I don't think `bootrr` tests are being run with Debian at all as the `baseline-nfs` jobs don't run it.  So adding this package here is a bit stale.  Adding it to `buildroot` is the important bit.

**Comment by gctucker** [comment]
We had issues creating new buildroot images as it's now being done with `kci_rootfs` and the changes didn't work initially.  Now this is all being resolved so hopefully all the PRs that depend on buildroot will also get resolved soon.

**Comment by nfraprado** [comment]
So, I was working on adding some kselftests and totally missed this PR, redoing the work of finding the missing configs. But since this PR is already here, I've rebased my work on top of it.

Since this PR was changed to only add the required kernel configs for adding support for TPM, it is standalone and doesn't require any other PR to get merged (ie it's not blocked). And the configs added in this PR would be useful even without merging the PRs related to the bootrr TPM test, since I've been working on #1025 and #1034 to add the tpm2 kselftest to some platforms and that test requires the configs added here. This also solves the question about the configs being useful upstream, since the tpm2 kselftest is part of the upstream kernel, so the changes in this PR are definitely useful for testing the upstream kernel.

Just to do a breakdown of the configs added here:

Needed for sona (#1034):

```
CONFIG_TCG_TIS_SPI=y
CONFIG_TCG_TIS_SPI_CR50=y
CONFIG_SPI=y
CONFIG_SPI_PXA2XX=y
```

Needed for grunt (#1025):

```
CONFIG_TCG_TIS_I2C_CR50=y
```

The following configs are needed for both, since they enable basic TPM operation. In theory we could get away without them because they're enabled by default in our current defconfig, but it's a good idea to explicitly enable them so that some future defconfig change doesn't break our TPM tests.

```
CONFIG_TCG_TPM=y
CONFIG_TCG_TIS=y
```

So I'd suggest we just merge this PR as is.

**Comment by gctucker** [comment]
The only thing, reading the discussion history, is that the TPM tests are being enabled in bootrr and that requires an updated buildroot image to confirm the config changes are working as expected. It just got side-tracked, now we have buildroot support in kci_rootfs so it's easier to create new images. It's something we should be able to all test and merge this week.

**Comment by gctucker** [comment]
Yes but if we enable these configs and the TPM tests still don't produce the expected results then there's a problem somewhere. That's why I'd like to have the results before merging this PR, at the same time as the bootrr ones.

**Pull request merged**